### ISSUE TITLE
fine-tune face recognition model

### ISSUE THREAD

**Issue by ncuxiaogang** [davisking/dlib#886]
@davisking 
Hi, I want to fine tune the face recognition model named 'dlib_face_recognition_resnet_model_v1.dat'. But I have no idea on how to do it. And I have not found some specific example on it. Would you please give an example on how to do it or give me some guidence about it? Thanks.

**Comment by davisking** [comment]
All you need to do is load the model and then you can run the dnn_trainer
like normal on the net.  There is no special step to perform fine tuning.
It's just like any other training.  Read the DNN introduction examples to
see how the API works.

**Comment by ncuxiaogang** [comment]
@davisking Thanks for your quick response. Again, I want to fine tune the face recognition model 'dlib_face_recognition_resnet_model_v1.dat' to classify a two classes problem. In my understanding, I should change the following code
// training network type
using net_type = loss_metric<fc_no_bias<128,avg_pool_everything<
                            level0<
                            level1<
                            level2<
                            level3<
                            level4<
                            max_pool<3,3,2,2,relu<bn_con<con<32,7,7,2,2,
                            input_rgb_image
                            >>>>>>>>>>>>;

to

// training network type
using net_type = loss_multiclass_log<fc<2, avg_pool_everything<
                            level0<
                            level1<
                            level2<
                            level3<
                            level4<
                            max_pool<3,3,2,2,relu<bn_con<con<32,7,7,2,2,
                            input_rgb_image
                            >>>>>>>>>>>>;

I have no idea whether the above is correct ? Thanks.

**Comment by ncuxiaogang** [comment]
@davisking To my understanding, the pair-loss is used in the face recognition model 'dlib_face_recognition_resnet_model_v1.dat'. But I want to use soft-max loss in the last layer to classify a two classes problems.

**Comment by davisking** [comment]
Definitely don't do that.  If you just want to make a classifier then train
a multi-class linear SVM or something like that on the 128D descriptor
vectors that come out of the neural network.  Don't mess with the network
definition at all.  Just run it to get the 128D vectors, then use them to
train a classifier.

**Comment by ncuxiaogang** [comment]
I have try to use the SVM to classify the 128D vectors. But the result is not very well. Because I want to classify the real face and fake face. If I have to change the dnn architecture to slove the problem above, how should I do. I just want to compare the results between the SVM classification and fine tune the face recognition model to classify a two classes problem ? Thanks for you quick response.

**Comment by yanchao12122** [comment]
it's not easy to classify the real face and face face, just using 2D images.
you should concern about 3D faces from faceID or kinects devices.

**Comment by ncuxiaogang** [comment]
@yanchao12122 But in mobile devices, I just use one camera. I can not get the depth information. Do you have any good idea to slove the problem? Thanks a lot.

**Comment by yanchao12122** [comment]
you can let the user do some special behaviors.
e.g. let him simle, open his mouse, or shake his head.

**Comment by ncuxiaogang** [comment]
@yanchao12122 I got it. Thanks
